File: P1 2019.1/Questao(1)/java/entities/Notebook.java
package entities;

public class Notebook extends Produto implements Eletronico {

    public Notebook(String id) {
        super(id);
    }

    public void ligar(){
        System.out.println("Ligando Notebook")
    }

    public void desligar(){
        System.out.println("Desligando Notebook")
    }
}

File: P1 2019.1/Questao(1)/python/entities/Notebook.py
from Eletronico import Eletronico
from Produto import Produto


class Notebook(Produto, Eletronico):
    def __init__(self, id_produto: str):
        super().__init__(id_produto)
        self.is_power_on = False

    def ligar(self) -> None:
        if not self.is_power_on:
            self.is_power_on = True
            print("Notebook Ligando")
        else:
            print("Notebook já está ligado")

    def desligar(self) -> None:
        if self.is_power_on:
            self.is_power_on = False
            print("Notebook Desligando")
        else:
            print("Notebook Ligando")
